From dbabad39123220898290166469d441a05fc9a481 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sat, 9 Dec 2023 20:13:05 +0100
Subject: [PATCH] util: Revert "Be more gracefull on the app_id"

This breaks telegram desktop with app-id org.telegram.desktop and a
desktop file of org.telegram.desktop.desktop.

We *only* made that change to not change the behaviour of the
end-session-dialog. All inhibitors there do have a proper app-id though
so let's just live without not appending '.desktop' there either.

This reverts commit 2a3763303b4305058ee0ff5f2b76b31016e80114.

Reported-by: Newbyte <newbie13xd@gmail.com>
Closes: https://gitlab.gnome.org/World/Phosh/phosh/-/issues/1010
---
 src/util.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/util.c b/src/util.c
index 87564b393..95879b7f5 100644
--- a/src/util.c
+++ b/src/util.c
@@ -77,12 +77,8 @@ phosh_get_desktop_app_info_for_app_id (const char *app_id)
     }
   }
 
-  if (!g_str_has_suffix (app_id, ".desktop"))
-    desktop_id = g_strdup_printf ("%s.desktop", app_id);
-  else
-    desktop_id = g_strdup (app_id);
-
-  g_assert (desktop_id);
+  desktop_id = g_strdup_printf ("%s.desktop", app_id);
+  g_return_val_if_fail (desktop_id, NULL);
   app_info = g_desktop_app_info_new (desktop_id);
 
   if (app_info)
-- 
GitLab

